### PR TITLE
Remove onLayoutMeasure and getLayoutSize

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #31540.

TODO:

- [x] Tests

### CODE REVIEW

File: extensions/amp-carousel/0.1/test/test-slidescroll.js [hunk 1]
```diff
@@ -22,25 +22,28 @@ import {
...
+import {installResizeObserverStub} from '../../../../testing/resize-observer-stub';
 import {user} from '../../../../src/log';
 
 describes.realWin(
   'SlideScroll',
   {
     amp: {
       extensions: ['amp-carousel'],
-      runtimeOn: true,
+      runtimeOn: false,
```
Comment (samouri) [inline extensions/amp-carousel/0.1/test/test-slidescroll.js:33]: nit: is this default value? if yes then we can remove

File: testing/resize-observer-stub.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      }
+    });
+  }
+}
+
+class ResizeObserverStub {
+  constructor(callback, onDisconnect) {
+    this.onDisconnect_ = onDisconnect;
+    this.callback = callback;
+    this.elements = [];
```
Comment (samouri) [inline testing/resize-observer-stub.js:61]: `Set`?

File: testing/resize-observer-stub.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+import {pushIfNotExist, removeItem} from '../src/utils/array';
+
+/**
+ * @param {!Object} sandbox
+ * @param {!Window} window
+ * @return {!ResizeObservers}
+ */
+export function installResizeObserverStub(sandbox, win) {
```
Comment (samouri) [inline testing/resize-observer-stub.js:24]: this file is a great idea! really need one for InOb too!

File: testing/resize-observer-stub.js (hunk 2 above)
Comment (dvoytenko) [reply testing/resize-observer-stub.js:61]: Good idea. done.

File: testing/resize-observer-stub.js (hunk 3 above)
Comment (dvoytenko) [reply testing/resize-observer-stub.js:24]: Yeah. I thought we'll try this pattern for these observers.

**Review by samouri — APPROVED** [review]

**Pull request merged**